types/react/index.d.ts

```diff
@@ -169,7 +169,7 @@ declare namespace React {
 
     interface ReactNodeArray extends Array<ReactNode> {}
     type ReactFragment = {} | ReactNodeArray;
-    type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined;
+    type ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null;
```

Comment:
If you're removing `undefined` from `ReactNode` you need to re-introduce it in `ReactNodeArray` (`extends Array<ReactNode | undefined>`) as `undefined` inside an array is valid.

Reply:
Good catch, done!